weave-js/src/util/date.ts

```diff
@@ -0,0 +1,236 @@
+        return result;
+      case 'month':
+        result.setMonth(result.getMonth() - amount);
+        return result;
+      case 'year':
+        result.setFullYear(result.getFullYear() - amount);
+        return result;
+    }
+  }
+
+  // Handle "in X days/weeks/months/years"
+  const inFuturePattern = /^in\s+(\d+)\s+(minute|hour|day|week|month|year)s?$/i;
+  const inFutureMatch = lowerStr.match(inFuturePattern);
+  if (inFutureMatch) {
+    const amount = parseInt(inFutureMatch[1], 10);
+    const unit = inFutureMatch[2].toLowerCase();
+    const result = new Date(now);
+
+    switch (unit) {
+      case 'minute':
+        result.setMinutes(result.getMinutes() + amount);
+        return result;
+      case 'hour':
+        result.setHours(result.getHours() + amount);
+        return result;
+      case 'day':
+        result.setDate(result.getDate() + amount);
+        return result;
+      case 'week':
+        result.setDate(result.getDate() + amount * 7);
+        return result;
+      case 'month':
+        result.setMonth(result.getMonth() + amount);
+        return result;
+      case 'year':
+        result.setFullYear(result.getFullYear() + amount);
+        return result;
+    }
+  }
+
+  // Handle last/next for basic time units
+  const lastNextPattern = /^(last|next)\s+(minute|hour|day|week|month|year)$/i;
+  const lastNextMatch = lowerStr.match(lastNextPattern);
+  if (lastNextMatch) {
+    const direction = lastNextMatch[1].toLowerCase();
+    const unit = lastNextMatch[2].toLowerCase();
+    const multiplier = direction === 'last' ? -1 : 1;
+    const result = new Date(now);
+
+    switch (unit) {
+      case 'minute':
+        result.setMinutes(result.getMinutes() + multiplier);
+        return result;
+      case 'hour':
+        result.setHours(result.getHours() + multiplier);
+        return result;
+      case 'day':
+        result.setDate(result.getDate() + multiplier);
+        return result;
+      case 'week':
+        result.setDate(result.getDate() + multiplier * 7);
+        return result;
+      case 'month':
+        result.setMonth(result.getMonth() + multiplier);
+        return result;
+      case 'year':
+        result.setFullYear(result.getFullYear() + multiplier);
+        return result;
+    }
+  }
+
+  // If all parsing attempts fail, return null
+  return null;
+};
+
+/**
+ * Determines if a date input string represents a relative date
+ *
+ * @param value The date string to check
+ * @returns True if the input represents a relative date, false otherwise
+ */
+export const isRelativeDate = (value: string): boolean => {
+  if (!value || typeof value !== 'string') {
+    return false;
+  }
+
+  const trimmedValue = value.trim().toLowerCase();
+
+  // Check for shorthand patterns (e.g., "1d", "2w", "3mo", "5h", "10m")
+  if (/^\d+([dwymoh]|mo)$/i.test(trimmedValue)) {
+    return true;
+  }
+
+  // Check for natural language relative terms
+  const relativeKeywords = [
+    'ago',
+    'last',
```

Comment:
I immediately tried typing "last friday" but saw that it didn't work

Reply:
hmm, fixing